app/uk/gov/hmrc/excisemovementcontrolsystemapi/repository/MovementRepository.scala

```diff
@@ -80,6 +80,13 @@ class MovementRepository @Inject()
       .map(_ => true)
   }
 
+  def getMovementById(id: String): Future[Option[Movement]] = {
```

Comment:
you may want to use headOption here:
`

def getMovementById(id: String): Future[Option[Movement]] = {
    collection.find(equal("_id", id)).headOption()
  }

`

Reply:
Updated :)